pkg/funky/server.go

```diff
@@ -0,0 +1,200 @@
+///////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+///////////////////////////////////////////////////////////////////////
+package funky
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+// Server an interface for managing function servers
+type Server interface {
+	GetPort() uint16
+	Invoke(input Message) (io.ReadCloser, error)
```

Comment:
This got "outdated", but still holds:

Both input and return types should be `*Message` reflecting the fact that servers accept and return the same message structure consisting of context and payload.

Reply:
Do we really want the language servers returning the whole message object? To keep the language servers as simple as possible I would think that they should respond with the function return payload only. We are capturing the rest of context in funky.

Reply:
How about errors returned by the function server? For example, we can't detect InputError in funky: it can only be detected by the function code.

Reply:
We're adding logs and detecting errors function server can't, that's true

Reply:
The function server throws a 400 BadRequest. I've translated all error handling in the function server to returning proper error codes.